### PR TITLE
sync up react_state's return type

### PR DESCRIPTION
**Pull request by zingale**
## PR summary

We had the return type as `bool`, but were returning an `int`.  This shouldn't matter,
but this makes it consistent.

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/reactions/Castro_react.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ Castro::do_old_reactions (Real time, Real dt)
     advance_status status {};
 
 #ifndef SIMPLIFIED_SDC
-    bool burn_success = true;
+    int burn_success{1};
```
Comment (maxpkatz) [inline Source/reactions/Castro_react.cpp:23]: We should also then change the `if (!burn_success)` checks

Comment (zingale) [reply Source/reactions/Castro_react.cpp:23]: fixed

File: Source/reactions/Castro_react.cpp [hunk 2]
```diff
@@ -843,11 +845,7 @@ Castro::react_state(Real time, Real dt)
 
     }
 
-    if (burn_success) {
-        return true;
-    } else {
-        return false;
-    }
+    return burn_success == 1;
```
Comment (maxpkatz) [inline Source/reactions/Castro_react.cpp:848]: Now that the return type is int, can we just return burn_success?

Comment (zingale) [reply Source/reactions/Castro_react.cpp:848]: yes

Comment (zingale) [reply Source/reactions/Castro_react.cpp]: updated

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**